### PR TITLE
Feature/pose estimation

### PR DESCRIPTION
**Pull request by stark710**

### CODE REVIEW

File: prog-template.c [hunk 1]
```diff
@@ -193,27 +198,43 @@ bool rgb_2_gray_scale(unsigned char* original, unsigned char* result) {
...
+    if(detections)
+    {
+
+        int i;
+        for (i = 0; i < zarray_size(detections); i++) {
+            zarray_get(detections, i, &det);
+            printf("detection %3d: id (%2dx%2d)-%-4d, hamming %d, margin %8.3f\n",
+                            i, det->family->nbits, det->family->h, det->id, det->hamming, det->decision_margin);
+            result = true;
+            if(result)
```
Comment (jsongCMU) [inline prog-template.c]: Don't need if statement here; result is set to true right before it

File: prog-template.c [hunk 2]
```diff
@@ -193,27 +198,43 @@ bool rgb_2_gray_scale(unsigned char* original, unsigned char* result) {
...
+                apriltag_detection_info_t info;
+                apriltag_pose_t pose;
+                info.det = det;
+                info.tagsize = TAG_SIZE;
+                info.fx = fx;
+                info.fy = fy;
+                info.cx = 0;
+                info.cy = 0;
+                double err = estimate_tag_pose(&info, &pose);
+                printf("Pose: Rotation matrix size: %3d X %3d Translation matrix size: %3d X %3d \n Rotation matrix: %lf %lf %lf \n %lf %lf %lf \n %lf %lf %lf \n Translation matrix: %lf %lf %lf \n",pose.R->nrows, pose.R->ncols, pose.t->nrows, pose.t->ncols,pose.R->data[0], pose.R->data[1], pose.R->data[2], pose.R->data[4], pose.R->data[4], pose.R->data[5], pose.R->data[6], pose.R->data[7], pose.R->data[8], pose.t->data[0], pose.t->data[1], pose.t->data[2]);
```
Comment (jsongCMU) [inline prog-template.c]: pose.R->data[4] printed twice; intentional?

File: prog-template.c (hunk 1 above)
Comment (indraneelpatil) [reply prog-template.c]: Done

File: prog-template.c (hunk 2 above)
Comment (indraneelpatil) [reply prog-template.c]: Done

**Review by stark710 — COMMENTED** [review]
Looks good. Great work stark!

**Pull request merged**